apps/trainers/forms.py

```diff
@@ -39,9 +44,24 @@ class Meta:
         fields = ['name', 'street', 'number', 'zipcode', 'city', 'country']
 
 
+def validate_upload(file):
+    content_type = magic.from_buffer(file.read(), mime=True)
+    print(content_type)
+    if not content_type in settings.CONTENT_TYPES:
+        raise forms.ValidationError(_(
+            'File type is not supported. Please upload only files of the following types: %s') % (
+                                        settings.CONTENT_TYPES))
+    if file.size > settings.MAX_UPLOAD_SIZE:
+        raise forms.ValidationError(
+            _('Please keep the filesize under %s. Current filesize: %s') % (
+                filesizeformat(settings.MAX_UPLOAD_SIZE),
+                filesizeformat(file.size)))
+
+
 class UploadForm(forms.ModelForm):
     title = forms.CharField(required=True, label=_('Title'))
-    url = forms.FileField(required=True)
+    url = forms.FileField(required=True, label='', validators=[validate_upload], )
```

Comment:
Hier könnten wir doch wieder einen Help Text anzeigen, in dem erklärt wird, welche File Types erlaubt sind und wie groß die sein sollen?

Reply:
Der kommt ja dann, wenn man ein falsches File hochlädt.

Reply:
Stimmt, aber ein bisschen mehr user-friendly wäre es schon, es vorher anzuzeigen 🤔 ?